src/WebJobs.Script.Scaling/AzureWorkerTable.cs

```diff
@@ -0,0 +1,230 @@
+            entity.ETag = "*";
+
+            try
+            {
+                var operation = TableOperation.Delete(entity);
+                await table.ExecuteAsync(operation);
+            }
+            catch (StorageException ex)
+            {
+                var webException = ex.InnerException as WebException;
+                if (webException != null)
+                {
+                    var response = webException.Response as HttpWebResponse;
+                    if (response != null && response.StatusCode == HttpStatusCode.NotFound)
+                    {
+                        return;
+                    }
+                }
+
+                throw;
+            }
+        }
+
+        public async Task<IEnumerable<IWorkerInfo>> List()
+        {
+            var query = new TableQuery<AzureWorkerInfo>();
+            query.Where(TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal, AzureEnvironments.WorkerPartitionKey));
+
+            var table = await EnsureWorkerTable();
+            var results = await ExecuteQueryAsync(table, query);
+            return results.OfType<IWorkerInfo>();
+        }
+
+        public async Task<IWorkerInfo> GetManager()
+        {
+            var table = await EnsureWorkerTable();
+
+            // read manager row
+            var operation = TableOperation.Retrieve<AzureWorkerInfo>(AzureEnvironments.ManagerPartitionKey, AzureEnvironments.ManagerRowKey);
+            var result = await table.ExecuteAsync(operation);
+            var manager = result.Result as IWorkerInfo;
+            if (manager != null)
+            {
+                // read from worker table
+                var partitionKey = AzureEnvironments.WorkerPartitionKey;
+                var rowKey = AzureEnvironments.GetWorkerRowKey(manager.StampName, manager.WorkerName);
+                operation = TableOperation.Retrieve<AzureWorkerInfo>(partitionKey, rowKey);
+                result = await table.ExecuteAsync(operation);
+                manager = result.Result as IWorkerInfo;
+            }
+
+            return manager;
+        }
+
+        public async Task SetManager(IWorkerInfo worker)
+        {
+            // update manager row
+            var entity = new AzureWorkerInfo
+            {
+                PartitionKey = AzureEnvironments.ManagerPartitionKey,
+                RowKey = AzureEnvironments.ManagerRowKey,
+                StampName = worker.StampName,
+                WorkerName = worker.WorkerName,
+                ETag = "*"
+            };
+
+            var operation = TableOperation.InsertOrReplace(entity);
+            var table = await EnsureWorkerTable();
+            await table.ExecuteAsync(operation);
+        }
+
+        private async Task<CloudTable> EnsureWorkerTable()
+        {
+            if (_workerTable == null)
+            {
+                var account = CloudStorageAccount.Parse(AzureEnvironments.StorageConnectionString);
```

Comment:
As discussed offline, the pre-conditions are Dynamic sku and AzureWebJobStorage connection string.   These whole thing are disabled if conditions are not met.